src/main/java/uk/gov/ons/ctp/response/casesvc/endpoint/CaseGroupEndpoint.java

```diff
@@ -154,4 +152,36 @@ private boolean isValidEvent(String event) {
     return Arrays.stream(CategoryName.values()).map(Enum::name).filter(e -> e.equals(event)).count()
         == 1;
   }
+
+  /**
+   * Deletes all casegroup data for a particular collection exercise casegroup FK REFERENCE
+   * constraint defines DELETE CASCADE on case case FK REFERENCE constraint defines DELETE CASCADE
+   * on caseevent
+   *
+   * @param collectionExerciseId The Collection Exercise UUID to delete for
+   * @return An appropriate HTTP repsonse code
+   */
+  @DeleteMapping("collectionExercises/{collectionExerciseId}")
+  public ResponseEntity<String> deleteCaseDataByCollectionExercise(
+      @PathVariable UUID collectionExerciseId) throws CTPException {
+
+    Long caseGroupCount =
+        caseGroupService.getAllCasesAgainstCollectionExerciseId(collectionExerciseId);
+    if (caseGroupCount == 0) {
+      log.with("collection_exercise_id", collectionExerciseId)
+          .info("No content found for Collection Exercise ID");
+      return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    }
+
+    log.with("collection_exercise_id", collectionExerciseId)
+        .info("Deleting cases {}", caseGroupCount);
```

Comment:
Is the `caseGroupCount` actually a `caseCount`. If we do this I think we should ignore counting the cases and simply count the casegroups (as that's what we're deleting). We know the database relationship integrity will take care of case and caseevent records.

However, regardless of that, do we need to do the above count at all, or simply use the response from the `deleteCaseGroupByCollectionExerciseId` request as to whether any data was deleted? (i.e. we should get a response of `0` from the deletion request). You could argue what's the point in making the delete request if there's no data to delete but it won't be much more expensive than doing a count up front.